### ISSUE TITLE
Customer's DataSets in BlockChain

### ISSUE THREAD

**Issue by markpyzhov** [MallCloud/contracts-api#11]
The original text of requirements is described in [wiki/BlockChain-communication](https://github.com/MallCloud/contracts-api/wiki/BlockChain-communication#business-requirements).

> 1.  All data, algorithms and ensembles (i.e, Analytic Objects) are registered on blockchain.

Can you explain "all data"? Customer's data will be registered either?
In what situations the blocks of Customer's Data will be used in Contracts API? As I suppose only in transactions of Notebook's execution (on paying royalties). Is that correct?

**Comment by ehillerbrand** [comment]
The easiest approach if you agree is that Customer, Data Scientist, and Canonical Data are all registered.  Only Data Scientist data would have transactions saved to chain because there is payment.

**Comment by ehillerbrand** [comment]
We would need a way of registering each field or the entire data source.

**Comment by markpyzhov** [comment]
> The easiest approach if you agree is that Customer, Data Scientist, and Canonical Data are all registered. Only Data Scientist data would have transactions saved to chain because there is payment.

Yes, easiest. I just need to understand all possible situations. When you say Business requirement I think that you expect some use cases. If so, can you share?

> We would need a way of registering each field or the entire data source.

Yes, created #16.

**Comment by ehillerbrand** [comment]
Here are the use cases

https://docs.google.com/document/d/1YKgzh0_mxFpm8wE95GQwv8lwSS61CcUEotzR0npF7CU/edit?usp=sharing
